spec/cli_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'spec_helper'
+require 'samwise'
+require 'json'
+require 'factory'
+require 'stringio'
+
+describe Samwise::Cli, vcr: { cassette_name: "Samwise::Cli", record: :new_episodes } do
+  let(:goodfile) {'spec/cli_mocks/test.json'}
+  let(:piped_file){$stdin = StringIO.new(File.read(goodfile))}
+  let(:badfile) {'spec/cli_mocks/bad.json'}
+  let(:user_array) {Factory.create_users(["1000", "100"], ["080037478", "080031478"])}
+  let(:abort_msg) {'please use a .json with a duns key'}
+
+	def output(method, file)
+    capture(:stdout){Samwise::Cli.new.invoke(method, [], {:infile => file})}
+  end
+
+  after(:example) do
+    $stdin = STDIN
+  end
+
+  describe ".verify" do
+    let(:method) {:verify}
+    let(:verified_user) {Factory.build_users_json(user_array, [{"verified":true}, {"verified":false}])+"\n"}
+    it "verifies the duns for each user with streamed infile" do
+        piped_file
+        expect(output(method, goodfile)).to eq(verified_user)
+    end
+    it "verifies the duns for each user when given an -i flagged test.json with duns key" do
+      expect(output(method, goodfile)).to eq(verified_user)
+    end
+
+    it "requests to recieve a good json when recieves json without duns key" do
+      begin
+        err = capture_stderr(output(method, badfile))
+      rescue Exception => e
+        expect(e.to_s).to eq(abort_msg)
+      end
+    end
+  end
+
+  describe ".excluded" do
+	  let(:method) {:excluded}
+	  let(:excluded_user){Factory.build_users_json(user_array, [{"excluded":false}, {"excluded":false}])+"\n"}
+	  it "checks if users are excluded for each user in a streamed infile" do
+	    piped_file
+	    expect(output(method, goodfile)).to eq(excluded_user)
+	  end
+
+	  it "checks if users are excluded when given an -i flagged test.json with duns key" do
+	      expect(output(method, goodfile)).to eq(excluded_user)
+	  end
+
+	  it "requests to recieve a good json when recieves json without duns key" do
+	    begin
+	      output(method, badfile)
+	    rescue Exception => e
+	      expect(e.to_s).to eq(abort_msg)
+	    end
+	  end
+  end
+
+  describe ".get_info" do
+	  let(:method){:get_info}
```

Comment:
still some spacing issues here and there. I'll start reviewing the substance of the code, but all these should be aligned before it's merged.